### ISSUE TITLE
Tooltip doesn't open automatically

### ISSUE THREAD

**Issue by wldcordeiro** [facelessuser/ColorHelper#30]
I'm using this plugin with the latest 3 dev build (3107) at time of writing, and while I can put my cursor in a color and right click to open the helper, it doesn't open automatically for me.

**Comment by facelessuser** [comment]
I am afraid there is just not enough info in this issue for me to help you.  Please see [bug reports](https://facelessuser.github.io/ColorHelper/contributing/#bug-reports) to learn what kind of info I am looking for.  I need you to be very specific so I can tell exactly why you are having trouble.  If I am not given enough info to reproduce, I will have to close this issue.

As far as I can see, ST3 build 3107 is working fine with ColorHelper when I use it.

**Comment by wldcordeiro** [comment]
OS: Ubuntu 15.10 64 Bit.


Debug mode is only printing this out in plain CSS files.

```
ColorHelper: Regions to search:
[]
```

**Comment by facelessuser** [comment]
Okay, let me pose some more questions:

- Post some example CSS you are using that isn't working.
- Are you using the default CSS syntax highlighter?
- Is it CSS found within an HTML document, and if so, what HTML syntax highlighter are you using?
- Have you tried to configure any of ColorHelpers settings, or is it stock?

Stuff like that.  I need to be able to reproduce your problem.  It is doubtful a linux specific issue.  It is most likely a syntax highlighter problem since ColorHelper uses the syntax in a file to determine when to popup.  I can't reproduce with what you've currently given me; you need to give actual CSS code and details that I can use to reproduce.

**Comment by wldcordeiro** [comment]
Just a note, watch your tone, you can come off condescending. The issue I assume is with code like this

```css
$pin-out: #DCC15B;
$master-icon: #23A2A4;
$shadow: rgba(0,0,0, 0.5);
$main: #191919;
$main-contrast: #FFF;
$secondary: #1E1E1E;
$tertiary: #323232;
$headers: #969696;
$inactive-icons: #323232;
$text-on-main: #5A5A5A;
$active: #D4D4D4;
$selection: #E2E2E2;
$primary: #6EACD4;
$alert: #E73130;
$window-close: #000;
```
I also use the `ColorHighlighter` plugin and it is able to resolve those colors even though its not official CSS code.

**Comment by facelessuser** [comment]
> Just a note, watch your tone, you can come off condescending.

I've said nothing condescending.  Keep in mind I am not getting paid for the plugin nor the support of the plugin.  This is done in my free time and this is not the only repo I provide and support for free.  I think I am being very patient and even trying to promptly address your problem.  I would be careful with accusations.

With that out of the way, let's move back to the problem at hand.

> I also use the ColorHighlighter plugin and it is able to resolve those colors even though its not official CSS code. So I thought this plugin would also see the hex codes and rgba as something to add a tooltip to.

Yes it can, but it must be configured to do so.  Currently, scanning for colors in the whole file to create a palette for the current file and auto popup are kind of tied together.  Special rules are given to only scan **valid** colors.  These rules are built around syntax so that only valid syntax blocks get scanned.  We ignore colors in comments, strings, etc.  We use that same logic to popup the dialog only when in valid colors as not to annoy the user.  In general, we cover most CSS, SASS, SCSS syntax highlighters out of the box.  But we might not cover all.

Some people prefer to just have the popup to show when the cursor is on it regardless of what file or syntax it is in.  If this is what you want, then this can be done by reconfiguring the rules to be extremely relaxed, but then ColorHelper scanner would be more relaxed as well and can end up scanning all of the file (even extremely large files), which could lead to poor performance on file save, edit, etc.  If you plan on relaxing the rules that much, the scanning to create a current file palette should be disabled.  There was a user who desired this and I covered what rule he should use instead in the forums.  But since the conversion to the new forum, some of that info has been corrupted a bit, so I will post an example here.  I am assuming a desire for CSS3 colors, so the rule will reflect that.  Just replace the existing rules with this and disable `enable_current_file_palette`:

```js
// Disable current file scanning to create palette of colors in current file.
"enable_current_file_palette": false,

// Scanning rules
"color_scanning": [
    {

        "syntax_files": [],
        "syntax_filter": "whitelist",
        "base_scopes": [],
        "scan_scopes": ["source", "text"],
        "allowed_colors": ["css3"],
        "extensions": [],
        "use_hex_argb": false,
        "compress_hex_output": true
    }
],
```

I believe that should get you what you want.  If you want the scanning of the current file enabled, we would want to use rules that are a bit tighter.  In that case I would need to know what language you posted was and what syntax highlighter you are using.

**Comment by wldcordeiro** [comment]
Thanks for your very detailed response and I'm sorry for seeming accusatory. I think it's just a matter of your curtness in the previous responses appearing to be condescending (at least to me.) Apologies if not. 

I used scope hunter to figure out the scopes within one of the colors in the snippet I posted above.

```
Scope:                         source.css
                               meta.selector.css
                               entity.other.attribute-name.id.css
```

**Comment by facelessuser** [comment]
I am glad you got it figured out.

I can be terse/blunt etc. at times (or as I like to call it efficient), but I am no Linus Torvalds: so there's that :smile:. I do try to not be rude, but I can't control how people react to me.  At times even playfulness in my response has sometimes been interpreted as rude as the joke may have been missed.  I'm not perfect and sometimes I snap at people, and usually I'll be the first to apologize when I realize it or am called out on it.  I also get that my personality doesn't always translate well in text form, but whacha gonna do; I am who I am.  I do apologize if my personality comes off as abrasive.

I try to give people prompt support, but their is no obligation on my side to provide this.  I am not running a business.  I get no compensation for my work, and at times it is thankless work.  At times I have to deal with issues like this: https://github.com/facelessuser/ApplySyntax/issues/81, and when I do, I die a little more on the inside :frown:.

I do have 2 pet peeves in regards to issues reports.  And while these things can bother me at times, I try not to blow up at people about it as I realize not everyone has the same experience when dealing with bug reports, but I will emphasize my expectations, and I do not apologize for it as I have outlined what I expect when requesting support in the [documentation](https://facelessuser.github.io/ColorHelper/contributing/#bug-reports).  My time is valuable to me, so I expect that people will attempt to read the documentation to answer their own questions, and I expect that people will help me help them and give me as much info as they can so I don't have to try and do psychic debugging or interrogate them :smile:.  I do understand though that some issues can be difficult to pin down for a user.

Hopefully there are no hard feelings as their was no malice in my responses: just trying to cut to the heart of the problem. Your third post clearly outlined you were not using CSS and outlined your expectations which I found immensely helpful and painted a more complete picture for me and in turn helped me to understand the problem.
